Framework/DataHandling/src/SaveOpenGenieAscii.cpp

```diff
@@ -46,340 +43,371 @@ void SaveOpenGenieAscii::init() {
 
-    if (type.std::string::find("vector") &&
-        type.std::string::find("double") != std::string::npos) {
+  // xunit & xlabel put in OpenGenie format
+  const std::string xunitsVal = "Time-of-Flight (\\\\gms)";
+  addToOutputBuffer("xunits", m_stringType, xunitsVal);
+  addToOutputBuffer("xlabel", m_stringType, xunitsVal);
 
-      auto tsp = ws->run().getTimeSeriesProperty<double>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
-    }
+  // yunit & ylabel put in OpenGenie format
+  const std::string yunitsVal = "Neutron counts / \\\\gms";
+  addToOutputBuffer("yunits", m_stringType, yunitsVal);
+  addToOutputBuffer("ylabel", m_stringType, yunitsVal);
+}
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("int") != std::string::npos) {
+/**
+  * Calculates the delta in the logged (i.e. not data) X Y Z values by
+  * from the minimum and maximum logged values and stores them in the
+  * output buffer
+  *
+  * @param unit :: The axis of this delta (e.g. 'X' or 'Y') as a string
+  * @param values :: Pointer to the time series log values to process
+  */
+void SaveOpenGenieAscii::calculateXYZDelta(const std::string &unit,
+                                           const Kernel::Property *values) {
+  // Cast to TimeSeries so we can use the min/max value methods
+  const auto positionValues =
+      static_cast<const TimeSeriesProperty<double> *>(values);
+
+  const double deltaValue =
+      positionValues->maxValue() - positionValues->minValue();
+
+  addToOutputBuffer('d' + unit, m_floatType, std::to_string(deltaValue));
+}
 
-      auto tsp = ws->run().getTimeSeriesProperty<int>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+/**
+  * Converts histogram X/Y/E data into a string representation
+  * to be written out and the number of data points
+  *
+  * @tparam histoData :: The histogram data to process
+  *
+  * @return :: An OpenGenie compatible string representation and
+  * the number of data points as packed as a tuple
+  */
+template <typename T>
+std::tuple<std::string, int>
+SaveOpenGenieAscii::convertWorkspaceData(const T &histoData) {
+  // Padding to apply after 10 data values
+  const std::string newLineStr = "\r\n    ";
+
+  // First 4 spaces for correct padding
+  std::string outputString("    ");
+  int valueCount(0);
+
+  // Working on primitive type so don't take ref
+  for (const auto val : histoData) {
+    if (valueCount % 10 == 0) {
+      outputString += newLineStr;
     }
+    valueCount++;
+    outputString += std::to_string(val) + ' ';
+  }
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("bool") != std::string::npos) {
-
-      auto tsp = ws->run().getTimeSeriesProperty<bool>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
-    }
+  return std::make_tuple(outputString, valueCount);
```

Comment:
Ahh originally it was returning three values (the type as well) so a tuple made sense at the time. I'll go and change this as I know a few places that can be turned into pairs